internal/validator/default/default_validator.go

```diff
@@ -33,20 +33,62 @@ func New(policy *bitcoin.Settings) validator.Validator {
 	}
 }
 
-func (v *DefaultValidator) ValidateTransaction(tx *bt.Tx, skipFeeValidation bool, skipScriptValidation bool) error { //nolint:funlen - mostly comments
-	//
-	// Each node will verify every transaction against a long checklist of criteria:
-	//
-	txSize := tx.Size()
-
-	// fmt.Println(hex.EncodeToString(tx.ExtendedBytes()))
-
+func (v *DefaultValidator) ValidateEFTransaction(tx *bt.Tx, skipFeeValidation bool, skipScriptValidation bool) error { //nolint:funlen - mostly comments
 	// 0) Check whether we have a complete transaction in extended format, with all input information
 	//    we cannot check the satoshi input, OP_RETURN is allowed 0 satoshis
 	if !v.IsExtended(tx) {
 		return validator.NewError(fmt.Errorf("transaction is not in extended format"), api.ErrStatusTxFormat)
 	}
 
+	// The rest of the validation steps
+	err := v.validateTransaction(tx, skipFeeValidation, skipScriptValidation)
+	if err != nil {
+		return err
+	}
+
+	// everything checks out
+	return nil
+}
+
+func (v *DefaultValidator) ValidateBeef(beefTx *beef.BEEF, skipFeeValidation, skipScriptValidation bool) error {
+	for _, btx := range beefTx.Transactions {
+		if !btx.IsMined() {
```

Comment:
You can save one indentations if you do check `IsMined()`.
I find fewer indentations usually a bit easier to read.

```
	for _, btx := range beefTx.Transactions {
		if btx.IsMined() {
			continue
		}
		
		tx := btx.Transaction

		// needs to be calculated here, because txs in beef are not in EF format
		if !skipFeeValidation {
			if err := v.validateBeefFees(tx, beefTx); err != nil {
				return err
			}
		}

		// needs to be calculated here, because txs in beef are not in EF format
		if !skipScriptValidation {
			if err := beef.ValidateScripts(tx, beefTx.Transactions); err != nil {
				return validator.NewError(err, api.ErrStatusUnlockingScripts)
			}
		}

		// purposefully skip the fee and scripts validation, because it's done above
		if err := v.validateTransaction(tx, true, true); err != nil {
			return err
		}
	}
```

Reply:
Sure, done